docs/data/material/customization/typography/TypographyCustomVariant.js

```diff
@@ -7,11 +7,35 @@ const theme = createTheme({
   typography: {
     // @ts-ignore
     poster: {
-      color: 'red',
+      fontSize: '4rem',
+      color: 'indianred',
     },
     // Disable v3 variant
     h3: undefined,
   },
+  components: {
+    MuiTypography: {
+      defaultProps: {
+        variantMapping: {
+          // we have to provide the default variantMapping first because currently
+          // custom entries will replace this whole object instead of being merged
+          h1: 'h1',
+          h2: 'h2',
+          h3: 'h3',
+          h4: 'h4',
+          h5: 'h5',
+          h6: 'h6',
+          subtitle1: 'h6',
+          subtitle2: 'h6',
+          body1: 'p',
+          body2: 'p',
+          inherit: 'p',
```

Comment:
```suggestion
```

After checking the implementation, these lines are not needed.

Here is the proof: https://codesandbox.io/s/naughty-ives-5vorcl?file=/demo.tsx

Reply:
Thanks for double checking ~ I see it falls back to the default variantMapping [here](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/Typography/Typography.js#L121)